File: tauri-template/src/lib/app-config.ts
/** Fork customization point — keep in sync with `src-tauri/src/config.rs`. */
export const appConfig = {
  appName: 'WebApp Template',
  identifier: 'com.lumotic.webapptemplate',
  allowedUrlPatterns: [] as string[],
  userAgentSuffix: 'WebAppTemplate-Tauri/1.0',
};

export function isUrlAllowed(url: string): boolean {
  if (appConfig.allowedUrlPatterns.length === 0) {
    return true;
  }
  return appConfig.allowedUrlPatterns.some((pattern) => url.startsWith(pattern));
}

File: flutter-template/lib/core/config/app_config.dart
/// Fork customization point — change these values when creating a new app.
class AppConfig {
  const AppConfig._();

  static const String appName = 'WebApp Template';
  static const String appId = 'com.lumotic.webapp_template';

  /// Optional URL scheme allowlist. Empty list allows any valid http/https URL.
  static const List<String> allowedUrlPatterns = [];

  static const String defaultUserAgentSuffix = 'WebAppTemplate/1.0';

  static bool isUrlAllowed(String url) {
    if (allowedUrlPatterns.isEmpty) {
      return true;
    }
    return allowedUrlPatterns.any(url.startsWith);
  }
}
